src/sentry/api/endpoints/group_ai_autofix.py

```diff
@@ -0,0 +1,170 @@
+        metadata = group.data.get("metadata", {})
+        metadata["autofix"] = {
+            "createdAt": created_at,
+            "status": "PROCESSING",
+        }
+
+        event_stacktrace = {"entries": latest_event.data.get("exception", {}).get("values", [])}
+        release_version = group.get_last_release()
+        if not release_version:
+            metadata["autofix"] = {
+                **metadata["autofix"],
+                "completedAt": datetime.now().isoformat(),
+                "status": "ERROR",
+                "fix": None,
+                "errorMessage": "Event has no release.",
+            }
+
+            group.data["metadata"] = metadata
+            group.save()
+
+            return HttpResponse(
+                status=400,
+            )
+
+        release: Release = Release.objects.get(version=release_version)
+        release_commits: list[ReleaseCommit] = ReleaseCommit.objects.filter(release=release)
+
+        commits: list[Commit] = [release_commit.commit for release_commit in release_commits]
+        base_commit: Commit | None = None
+        for commit in commits:
+            repo: Repository = Repository.objects.get(id=commit.repository_id)
+            # Hardcoded to only accept getsentry/sentry repo for now
+            if repo.external_id == "getsentry/sentry":
+                base_commit = commit
+                break
+
+        if not base_commit:
+            metadata["autofix"] = {
+                **metadata["autofix"],
+                "completedAt": datetime.now().isoformat(),
+                "status": "ERROR",
+                "fix": None,
+                # Hardcoded to only accept getsentry/sentry repo for now
+                "errorMessage": "No valid base commit found for release; only getsentry/sentry repo is supported right now.",
+            }
+            group.data["metadata"] = metadata
+            group.save()
+
+            return HttpResponse(
+                status=400,
+            )
+
+        try:
+            seer_connection_pool.urlopen(
+                "POST",
+                "/v0/automation/autofix",
+                body=json.dumps(
+                    {
+                        "base_commit_sha": base_commit.key,
+                        "issue": {
+                            "id": str(group.id),
+                            "title": group.title,
+                            "events": [event_stacktrace],
+                        },
+                        "additional_context": data.get("additional_context", ""),
+                    }
+                ),
+                headers={"content-type": "application/json;charset=utf-8"},
+            )
+
+            # Mark the task as completed after TIMEOUT_SECONDS
+            check_for_timeout.apply_async(
+                kwargs={
+                    "group_id": group.id,
+                    "created_at": created_at,
+                },
+                countdown=TIMEOUT_SECONDS,
+            )
```

Comment:
nit: IMO this should be in `else:` part of `try/except`

Reply:
Do you mean in the `except:` portion? But the timeout countdown should be started when there is a successful call right because we're having it catch the edge case of seer timing out and not updating the status in the RPC endpoint:

https://github.com/getsentry/sentry/blob/6dffc8bc4ee9db8723ba186db0a98ac6d9be3e3d/src/sentry/api/endpoints/seer_rpc.py#L144-L179